lib/Parse/ParseDecl.cpp

```diff
@@ -3547,13 +3553,17 @@ parseDeclTypeAlias(Parser::ParseDeclOptions Flags, DeclAttributes &Attributes) {
 
     UnderlyingTy = parseType(diag::expected_type_in_typealias);
     Status |= UnderlyingTy;
-    if (UnderlyingTy.isNull())
+    if (UnderlyingTy.isNull()) {
+      // If there is an attempt to do code completion
+      // inside of typealias type, let's set generic
+      // parameters we have already seen, since underlying
+      // type might need this information.
+      if (UnderlyingTy.hasCodeCompletion())
+        TAD->setGenericParams(genericParams);
       return Status;
```

Comment:
Must we early-exit here? It seems like we'd be better off continuing, because the change to `TypeCheckDecl` will make it okay to have a null underlying type?

Reply:
We can continue but I'd have to do a check for `if (UnderlyingTy.hasCodeCompletion())` later on because otherwise it's going to error with diagnostic about `expected '=' in type alias declaration`, I just thought that it might be better to preserve current flow

Reply:
but let me change it.